webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeGroupRepository.scala

```diff
@@ -829,23 +836,26 @@ class WoLDAPNodeGroupRepository(
 
     groupLibMutex.readLock(for {
       con          <- ldap
-      oldParents   <- if(autoExportOnModify) {
-                        (for {
-                          parent   <- getNodeGroupCategory(nodeGroupId)
-                          parents  <- getParents_NodeGroupCategory(parent.id)
-                        } yield (parent::parents).map( _.id ))
-                      } else Nil.succeed
-      existing     <- getSGEntry(con, nodeGroupId).notOptional("Error when trying to check for existence of group with id %s. Can not update".format(nodeGroupId.value))
-      oldGroup     <- mapper.entry2NodeGroup(existing).toIO.chainError("Error when trying to get the existing group with id %s".format(nodeGroupId.value))
-      systemCheck  <- if(oldGroup.isSystem) "You can not move system group".fail else UIO.unit
-
-      groupRDN     <- existing.rdn.notOptional("Error when retrieving RDN for an exising group - seems like a bug")
-      name         <- checkNameAlreadyInUse(con, oldGroup.name, nodeGroupId)
-      exists       <- ZIO.when(name) {
-                        s"Cannot change the group name to ${oldGroup.name}: there is already a group with the same name".fail
-                      }
-      newParentDn  <- getContainerDn(con, containerId).chainError(s"Couldn't find the new parent category when updating group ${oldGroup.name}")
-      result       <- groupLibMutex.writeLock { con.move(existing.dn, newParentDn) }
+      res          <- groupLibMutex.writeLock { for {
```

Comment:
I think their should not be a readLock up there, but rather a writeLock and so remove this one

Reply:
clearly